pkg/virt-operator/creation/csv/csv.go

```diff
@@ -0,0 +1,350 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+package csv
+
+import (
+	"encoding/json"
+
+	"github.com/coreos/go-semver/semver"
+	csvv1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/pkg/virt-operator/creation/components"
+	"kubevirt.io/kubevirt/pkg/virt-operator/creation/rbac"
+)
+
+type NewClusterServiceVersionData struct {
+	Namespace            string
+	KubeVirtVersion      string
+	OperatorImageVersion string
+	DockerPrefix         string
+	ImagePullPolicy      string
+	Verbosity            string
+	CsvVersion           string
+	VirtApiSha           string
+	VirtControllerSha    string
+	VirtHandlerSha       string
+	VirtLauncherSha      string
+	Replicas             int
+	IconBase64           string
+	ReplacesCsvVersion   string
+	CreatedAtTimestamp   string
+}
+
+type csvClusterPermissions struct {
+	ServiceAccountName string              `json:"serviceAccountName"`
+	Rules              []rbacv1.PolicyRule `json:"rules"`
+}
+type csvDeployments struct {
+	Name string                `json:"name"`
+	Spec appsv1.DeploymentSpec `json:"spec,omitempty"`
+}
+
+type csvStrategySpec struct {
+	ClusterPermissions []csvClusterPermissions `json:"clusterPermissions"`
+	Deployments        []csvDeployments        `json:"deployments"`
+}
+
+var description = `
+**KubeVirt** is a virtual machine management add-on for Kubernetes.
+The aim is to provide a common ground for virtualization solutions on top of
+Kubernetes.
+
+# Virtualization extension for Kubernetes
+
+At its core, KubeVirt extends [Kubernetes](https://kubernetes.io) by adding
+additional virtualization resource types (especially the 'VirtualMachine' type) through
```

Comment:
would be nice to keep the backticks here around VirtualMachine IMHO, because the description is rendered as markdown. Backticks can't be escaped, but you can do ....\` + "\`VirtualMachine\`" + \`....

Reply:
AH, yes. good idea. I'll give that a shot